app/public/src/pages/lobby.tsx

```diff
@@ -242,25 +242,25 @@ export async function joinLobbyRoom(
         dispatch(logIn(user))
         try {
           const token = await user.getIdToken()
-
-          const reconnectToken: string = localStore.get(LocalStoreKeys.RECONNECTION_TOKEN)
+          let lobby: Room<ICustomLobbyState> | undefined = store.getState().network.lobby
+          if (lobby?.connection.isOpen) {
+            localStore.set(LocalStoreKeys.RECONNECTION_LOBBY, lobby.reconnectionToken, 30)
+            await lobby.leave(false)
+            lobby = undefined
+          }
+          const reconnectToken: string = localStore.get(LocalStoreKeys.RECONNECTION_LOBBY)
           if (reconnectToken) {
             try {
-              const lobbyRoom: Room<ICustomLobbyState> = await client.reconnect(reconnectToken)
-              if (lobbyRoom.name === "lobby") {
-                dispatch(joinLobby(lobbyRoom))
-              } else if (lobbyRoom.connection.isOpen) {
-                lobbyRoom.connection.close()
-              }
+              lobby = await client.reconnect(reconnectToken)
             } catch (error) {
-              logger.log(error)
-              localStore.delete(LocalStoreKeys.RECONNECTION_TOKEN)
+              localStore.delete(LocalStoreKeys.RECONNECTION_LOBBY)
             }
           }
-          const lobby = store.getState().network.lobby
           const room: Room<ICustomLobbyState> = lobby ?? await client.join("lobby", {
             idToken: token
           })
+          // store key for 5 minutes, but can still only rejoin within 30 seconds of leaving
```

Comment:
can you explaining the reasoning about the 5 minutes client side token expiration ?

Reply:
Currently we have a cron job that runs every minute and kick all players that habe been in lobby room for more than one minute. So this expiration time doesn't seem very useful at the moment. It's not really a problem, just I could see it much longer, anyway server will be authrotitative when it comes to kicking inactive players

Reply:
2 minutes max, if you're desynced with the cron job tick. But as I said, doesn't change much, it could be one hour or one day